File: Python/BOJ/문자열/1543.py
//
//  1543.py
//  Algorithm
//
//  Created by 유정주 on 2023/06/16.
//

import sys; readLine = sys.stdin.readline

s = readLine().rstrip()
target = readLine().rstrip()
length = len(target)

result = 0
i = 0
while i <= len(s) - length:
    if s[i:i+length] == target:
        result += 1
        i += length
        continue

    i += 1

print(result)

File: Swift/BOJ/1000/1543.swift
//
//  1543.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/16.
//

import Foundation

let s = Array(readLine()!).map { String($0) }
let word = readLine()!

var result = 0
var i = 0
while i <= s.count - word.count {
    if s[i..<i + word.count].joined() == word {
        result += 1
        i += word.count
        continue
    }
    
    i += 1
}

print(result)
